Compiler Error with features sqlx-postgres and proxy

<!--

Welcome! Thank you for reporting bugs!

First of all, please star our repo. Your support is vital to the continued maintenance of SeaORM.

Want to ask a question? You can reach us via:

- Discord: https://discord.com/invite/uCPdDXzbdv
- GitHub Discussions: https://github.com/SeaQL/sea-orm/discussions/new

Please make sure that you are not asking for a missing feature; a bug is incorrect behavior -
either in the feature specification or implementation. Feature requests should be first raised in discussions.

Please also make sure your description is clear and precise - maintainers don't have access to your
code and can't see what you have seen. Please avoid vague descriptions like "they are different"
or "the program crashes" - in either case, provide exact information.

If you are certain there is a bug, please provide a reproducible example, which helps the investigator
to pin-point the bug and the implementor to verify that a solution is satisfactory. Bug reports without
reproducible example may be closed or dangle forever.

Finally, please search for existing issues and discussions before submission. Feel free to revive old
threads if you have new information to add, but please don't ask for ETA or "+1".

-->

## Description

```
$ git checkout 1.1.19
$ cargo build --no-default-features --features proxy,sqlx-postgres
   Compiling sea-orm v1.1.19 (/home/msrd0/git/sea-orm)
warning: unexpected `cfg` condition value: `with-mac_address`
   --> src/driver/sqlx_postgres.rs:653:31
    |
653 |                         #[cfg(feature = "with-mac_address")]
    |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: expected values for `feature` are: `bigdecimal`, `chrono`, `debug-print`, `default`, `ipnetwork`, `json-array`, `macros`, `mariadb-use-returning`, `mock`, `pgvector`, `postgres-array`, `postgres-vector`, `proxy`, `runtime-actix`, `runtime-actix-native-tls`, `runtime-actix-rustls`, `runtime-async-std`, `runtime-async-std-native-tls`, `runtime-async-std-rustls`, `runtime-tokio`, `runtime-tokio-native-tls`, `runtime-tokio-rustls`, `rust_decimal`, `sea-orm-internal`, `sea-query-binder`, `seaography`, `serde_json`, `sqlite-use-returning-for-3_35`, `sqlx`, `sqlx-all`, `sqlx-dep`, `sqlx-mysql`, `sqlx-postgres`, `sqlx-sqlite`, and `tests-cfg` and 9 more
    = help: consider adding `with-mac_address` as a feature in `Cargo.toml`
    = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration
    = note: `#[warn(unexpected_cfgs)]` on by default

warning: unexpected `cfg` condition value: `with-mac_address`
   --> src/driver/sqlx_postgres.rs:659:35
    |
659 |                         #[cfg(all(feature = "with-mac_address", feature = "postgres-array"))]
    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: expected values for `feature` are: `bigdecimal`, `chrono`, `debug-print`, `default`, `ipnetwork`, `json-array`, `macros`, `mariadb-use-returning`, `mock`, `pgvector`, `postgres-array`, `postgres-vector`, `proxy`, `runtime-actix`, `runtime-actix-native-tls`, `runtime-actix-rustls`, `runtime-async-std`, `runtime-async-std-native-tls`, `runtime-async-std-rustls`, `runtime-tokio`, `runtime-tokio-native-tls`, `runtime-tokio-rustls`, `rust_decimal`, `sea-orm-internal`, `sea-query-binder`, `seaography`, `serde_json`, `sqlite-use-returning-for-3_35`, `sqlx`, `sqlx-all`, `sqlx-dep`, `sqlx-mysql`, `sqlx-postgres`, `sqlx-sqlite`, and `tests-cfg` and 9 more
    = help: consider adding `with-mac_address` as a feature in `Cargo.toml`
    = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration

error[E0599]: no variant or associated item named `Json` found for enum `sea_query::Value` in the current scope
   --> src/driver/sqlx_postgres.rs:614:52
    |
614 |                         "JSON" | "JSONB" => Value::Json(
    |                                                    ^^^^ variant or associated item not found in `sea_query::Value`

error[E0599]: no variant or associated item named `Json` found for enum `ArrayType` in the current scope
   --> src/driver/sqlx_postgres.rs:621:51
    |
621 | ...                   sea_query::ArrayType::Json,
    |                                             ^^^^ variant or associated item not found in `ArrayType`

error[E0599]: no variant or associated item named `Json` found for enum `sea_query::Value` in the current scope
   --> src/driver/sqlx_postgres.rs:627:63
    |
627 | ...                   .map(|val| Value::Json(Some(Box::new(val))))
    |                                         ^^^^ variant or associated item not found in `sea_query::Value`

warning: value assigned to `self` is never read
   --> src/database/transaction.rs:152:9
    |
152 |         self.open = false;
    |         ^^^^^^^^^^^^^^^^^
    |
    = help: maybe it is overwritten before being read?
    = note: `#[warn(unused_assignments)]` (part of `#[warn(unused)]`) on by default

warning: value assigned to `self` is never read
   --> src/database/transaction.rs:192:9
    |
192 |         self.open = false;
    |         ^^^^^^^^^^^^^^^^^
    |
    = help: maybe it is overwritten before being read?

For more information about this error, try `rustc --explain E0599`.
warning: `sea-orm` (lib) generated 4 warnings
error: could not compile `sea-orm` (lib) due to 3 previous errors; 4 warnings emitted
```

## Steps to Reproduce

see above

### Expected Behavior

no error

### Actual Behavior

error

### Reproduces How Often

as often as attempted

### Workarounds

enable with-json feature eventhough it is not needed

## Versions

<!-- You can get this information from the output of `cargo tree | grep sea-` from the console. Also, please include the database and OS that you are running. -->
git tag 1.1.19 of this very repository

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -51,6 +51,7 @@ inventory = { version = "0.3", optional = true }
 ipnetwork = { version = "0.20", default-features = false, optional = true }
 itertools = "0.14.0"
 log = { version = "0.4", default-features = false }
+mac_address = { version = "1.1", default-features = false, optional = true }
 ouroboros = { version = "0.18", default-features = false }
 pgvector = { version = "~0.4", default-features = false, optional = true }
 rust_decimal = { version = "1", default-features = false, features = [
@@ -134,7 +135,7 @@ postgres-vector = [
     "sea-query/postgres-vector",
     "sea-query-sqlx?/postgres-vector",
 ]
-proxy = ["serde_json", "serde/derive"]
+proxy = ["serde/derive"]
 rbac = ["sea-query/audit", "macros"]
 runtime-async-std = ["sqlx?/runtime-async-std"]
 runtime-async-std-native-tls = [
@@ -211,8 +212,14 @@ with-json = [
     "bigdecimal?/serde",
     "uuid?/serde",
     "time?/serde",
+    "mac_address?/serde",
     "pgvector?/serde",
     "sea-query-sqlx?/with-json",
+]
+with-mac_address = [
+    "mac_address",
+    "sea-query/with-mac_address",
+    "sea-query-sqlx?/with-mac_address",
 ]
 with-rust_decimal = [
     "rust_decimal",

--- a/src/driver/sqlx_postgres.rs
+++ b/src/driver/sqlx_postgres.rs
@@ -626,12 +626,16 @@ pub(crate) fn from_sqlx_postgres_row_to_proxy_row(row: &sqlx::postgres::PgRow) -
                                 }),
                         ),
 
+                        #[cfg(feature = "with-json")]
                         "JSON" | "JSONB" => Value::Json(
                             row.try_get::<Option<serde_json::Value>, _>(c.ordinal())
                                 .expect("Failed to get json")
                                 .map(Box::new),
                         ),
-                        #[cfg(any(feature = "json-array", feature = "postgres-array"))]
+                        #[cfg(all(
+                            feature = "with-json",
+                            any(feature = "json-array", feature = "postgres-array")
+                        ))]
                         "JSON[]" | "JSONB[]" => Value::Array(
                             sea_query::ArrayType::Json,
                             row.try_get::<Option<Vec<serde_json::Value>>, _>(c.ordinal())

--- a/src/dynamic/model.rs
+++ b/src/dynamic/model.rs
@@ -135,6 +135,9 @@ fn try_get(res: &QueryResult, pre: &str, col: &str, ty: &ArrayType) -> Result<Va
 
         #[cfg(feature = "with-ipnetwork")]
         ArrayType::IpNetwork => Value::IpNetwork(res.try_get(pre, col)?),
+
+        #[cfg(feature = "with-mac_address")]
+        ArrayType::MacAddress => Value::MacAddress(res.try_get(pre, col)?),
     })
 }
 

--- a/src/executor/query.rs
+++ b/src/executor/query.rs
@@ -877,6 +877,9 @@ try_getable_uuid!(uuid::fmt::Urn, uuid::Uuid::urn);
 #[cfg(feature = "with-ipnetwork")]
 try_getable_postgres!(ipnetwork::IpNetwork);
 
+#[cfg(feature = "with-mac_address")]
+try_getable_postgres!(mac_address::MacAddress);
+
 impl TryGetable for u32 {
     #[allow(unused_variables)]
     fn try_get_by<I: ColIdx>(res: &QueryResult, idx: I) -> Result<Self, TryGetError> {
@@ -1096,6 +1099,9 @@ mod postgres_array {
 
     #[cfg(feature = "with-ipnetwork")]
     try_getable_postgres_array!(ipnetwork::IpNetwork);
+
+    #[cfg(feature = "with-mac_address")]
+    try_getable_postgres_array!(mac_address::MacAddress);
 
     #[allow(unused_macros)]
     macro_rules! try_getable_postgres_array_uuid {
@@ -1647,6 +1653,9 @@ try_from_u64_err!(uuid::Uuid);
 
 #[cfg(feature = "with-ipnetwork")]
 try_from_u64_err!(ipnetwork::IpNetwork);
+
+#[cfg(feature = "with-mac_address")]
+try_from_u64_err!(mac_address::MacAddress);
 
 #[cfg(test)]
 mod tests {

